src/ChainSafe.Gaming.Lootboxes.Chainlink/ILootboxService.cs

```diff
@@ -76,14 +96,14 @@ public interface ILootboxService
         /// Initiates the process for the current user to claim their rewards.
         /// </summary>
         /// <returns>An instance of <see cref="LootboxRewards"/> containing the details of the claimed rewards.</returns>
-        Task<LootboxRewards> ClaimRewards();
+        Task ClaimRewards();
 
         /// <summary>
         /// Initiates the process for a specified user to claim their rewards.
         /// </summary>
         /// <param name="account">User's public address from which rewards are to be claimed.</param>
         /// <returns>An instance of <see cref="LootboxRewards"/> containing the details of the claimed rewards for the specified user.</returns>
-        Task<LootboxRewards> ClaimRewards(string account);
+        Task ClaimRewards(string account);
```

Comment:
Think because we've went with our event manager instead of doing the events inside of the transaction itself, so that we're fully utilizing the SDK.

Reply:
because it just uses send now and invokes an event rather than scanning the receipt for better UX.

Claim rewards contract send is exectued like so:

```
public async Task ClaimRewards(string account)
{
    await this.contract.Send("claimRewards", new object[] { account });
}
```

This leads to the RewardsClaimedEvent emitted by the contract as we've subscribed to the event in the lifecycle.

```
await eventManager.Subscribe<RewardsClaimedEvent>(ExtractRewards, contractAddress);
```

It then sorts through your modified method and deserializes the response and invokes the OnRewardsClaimedMethod so the reward menu pops up.
```
private void ExtractRewards(RewardsClaimedEvent rewardsClaimedEvent)
        {
            var rewards = LootboxRewards.Empty;
            var rewardType = this.rewardTypeByTokenAddress[rewardsClaimedEvent.TokenAddress];
            switch (rewardType)
            {
                case RewardType.Erc20:
                    rewards.Erc20Rewards.Add(new Erc20Reward
                    {
                        ContractAddress = rewardsClaimedEvent.TokenAddress,
                        AmountRaw = rewardsClaimedEvent.Amount,
                    });
                    break;
                case RewardType.Erc721:
                    rewards.Erc721Rewards.Add(new Erc721Reward
                    {
                        ContractAddress = rewardsClaimedEvent.TokenAddress,
                        TokenId = rewardsClaimedEvent.TokenId,
                    });
                    break;
                case RewardType.Erc1155:
                    rewards.Erc1155Rewards.Add(new Erc1155Reward
                    {
                        ContractAddress = rewardsClaimedEvent.TokenAddress,
                        TokenId = rewardsClaimedEvent.TokenId,
                        Amount = rewardsClaimedEvent.Amount,
                    });
                    break;
                case RewardType.Erc1155Nft:
                    rewards.Erc1155NftRewards.Add(new Erc1155NftReward
                    {
                        ContractAddress = rewardsClaimedEvent.TokenAddress,
                        TokenId = rewardsClaimedEvent.TokenId,
                    });
                    break;
                case RewardType.Unset:
                default:
                    throw new ArgumentOutOfRangeException();
            }

            OnRewardsClaimed?.Invoke(rewards);
        }
```